### DISCUSSION TITLE
[ENG-180] Support Integration with Deno Deploy

### DISCUSSION THREAD

**Discussion by adoublef** [Infisical/infisical#5431] [Feature Request]
### Feature description

Deno Deploy is a cloud provider for hosting Deno projects. It is possible to update the env using their API [as seen here](https://github.com/skymethod/denoflare/blob/master/common/deploy/deno_deploy_api.ts#L36). The full API endpoint looks like

```
PATCH https://dash.deno.com/api/projects/${projectId}/env
```

With the response body being `Record<EnvName, string| null>` There is not official documentation for this yet though.

### Why would it be useful?

Deno Deploy is a popular place to deploy applications and so streamlines secret sharing just like the integrations for [*Fly.io*](http://Fly.io)*, Vercel, AWS and more*.

### Additional context

There is no official documentation on the feature currently. This infomation was shared within the Official Deno server

<sub>From [SyncLinear.com](https://synclinear.com) | [ENG-180](https://linear.app/infisical/issue/ENG-180/support-integration-with-deno-deploy)</sub>

**Comment by adoublef** [comment]
I noticed that the backend is written in Typescript which is great as there is [this api helper](https://github.com/denoland/deployctl/blob/main/src/utils/api.ts) so could help with creating the integration

**Comment by adoublef** [comment]
I'd like to take this one please

**Comment by akhilmhdh** [comment]
Moving to feature request discussions
